static/js/timerender.js

```diff
@@ -164,6 +164,51 @@ exports.render_date = function (time, time_above, today) {
     return node;
 };
 
+// Renders the timestamp returned by the !time() markdown syntax.
+exports.render_markdown_timestamp = function (time, now, text) {
+    now = now || moment();
+    let timestring = time.format('ddd, MMM D');
+    if (now.year() !== time.year()) {
+        timestring += time.format(' YYYY');
+    }
+    timestring += ',';
+    const military_time = page_params.twenty_four_hour_time;
+    if (military_time) {
+        timestring += time.format(' HH');
+    } else {
+        timestring += time.format(' h');
+    }
+    if (military_time || time.minutes() !== 0 || time.seconds() !== 0) {
+        timestring += time.format(':mm');
+    }
+    if (time.seconds() !== 0) {
+        timestring += time.format(":ss");
+    }
+    if (!military_time) {
+        timestring += time.format(" A");
+    }
+    const titlestring = "This time is in your timezone. Original text was '" + text + "'.";
+    return {
+        text: timestring,
+        title: titlestring,
+    };
+};
+
+exports.replace_markdown_timestamp = (elem) => {
+    // Populate each timestamp span with mentioned time
+    // in user's local timezone.
+    const timestamp = moment.unix($(elem).attr('data-timestamp'));
+    if (timestamp.isValid() && $(elem).attr('data-timestamp') !== null) {
+        const text = $(elem).text();
+        const rendered_time = exports.render_markdown_timestamp(timestamp, null, text);
+        $(elem).text(rendered_time.text);
+        $(elem).attr('title', rendered_time.title);
+    } else {
+        elem.classList.remove('timestamp');
+        elem.setAttribute('title', 'Could not parse timestamp.');
+    }
+};
```

Comment:
This is a function we want to call on all `rendered_markdown` elements we receive from the server, right?  I think maybe we should do a little PR that extracts the logic we have in `post_process` in `message_list_view.js` to live in a `rendered_markdown.js` that we can call from multiple places, and this belongs there.

Reply:
Fixed in #15049.